### ISSUE TITLE
G112 is invalid if ReadTimeout is set

### ISSUE THREAD

**Issue by ainar-g** [securego/gosec#823]
### Summary 

G112 should not be triggered if `http.Server.ReadTimeout` is set while `http.Server.ReadHeaderTimeout` isn't.

### Steps to reproduce the behavior

```go
package main

import (
	"fmt"
	"net/http"
	"time"
)

func main() {
	fmt.Println(&http.Server{
		ReadTimeout:  1 * time.Second,
		WriteTimeout: 1 * time.Second,
		IdleTimeout:  1 * time.Second,
	})
}
```

### gosec version

```none
/home/ainar/go/bin/gosec: go1.18.3
        path    github.com/securego/gosec/v2/cmd/gosec
        mod     github.com/securego/gosec/v2    v2.12.0 h1:CQWdW7ATFpvLSohMVsajscfyHJ5rsGmEXmsNcsDNmAg=
        dep     github.com/google/uuid  v1.3.0  h1:t6JiXgmwXMjEs8VusXIJk2BXHsn+wx8BZdTaoZ5fu7I=
        dep     github.com/gookit/color v1.5.1  h1:Vjg2VEcdHpwq+oY63s/ksHrgJYCTo0bwWvmmYWdE9fQ=
        dep     github.com/nbutton23/zxcvbn-go  v0.0.0-20210217022336-fa2cb2858354      h1:4kuARK6Y6FxaNu/BnU2OAaLF86eTVhP2hjTB6iMvItA=
        dep     github.com/xo/terminfo  v0.0.0-20210125001918-ca9a967f8778      h1:QldyIu/L63oPpyvQmHgvgickp1Yw510KJOqX7H24mg8=
        dep     golang.org/x/mod        v0.6.0-dev.0.20220419223038-86c51ed26bb4        h1:6zppjxzCulZykYSLyVDYbneBfbaBIQPYMevg0bEwv2s=
        dep     golang.org/x/sys        v0.0.0-20220319134239-a9b59b0215f8      h1:OH54vjqzRWmbJ62fjuhxy7AxFFgoHN0/DPc/UrL8cAs=
        dep     golang.org/x/tools      v0.1.11 h1:loJ25fNOEhSXfHrpoGj91eCUThwdNX6u24rO1xnNteY=
        dep     gopkg.in/yaml.v2        v2.4.0  h1:D8xgwECY7CYvx+Y2n4sBz93Jn9JRvxdiyyo8CTfuKaY=
        build   -compiler=gc
        build   CGO_ENABLED=1
        build   CGO_CFLAGS=
        build   CGO_CPPFLAGS=
        build   CGO_CXXFLAGS=
        build   CGO_LDFLAGS=
        build   GOARCH=amd64
        build   GOOS=linux
        build   GOAMD64=v1
```

### Go version (output of 'go version')

```none
go version go1.18.3 linux/amd64
```

### Operating system / Environment

```none
Ubuntu 20.04.4 LTS
```

### Expected behavior

No error, because:

```sh
go doc http.Server.ReadHeaderTimeout
```

<blockquote>

<pre>
package http // import "net/http"

type Server struct {
    // ReadHeaderTimeout is the amount of time allowed to read request headers. The
    // connection's read deadline is reset after reading the headers and the
    // Handler can decide what is considered too slow for the body. <strong>If
    // ReadHeaderTimeout is zero, the value of ReadTimeout is used.</strong> If both are
    // zero, there is no timeout.
    ReadHeaderTimeout time.Duration

    // ... other fields elided ...
}
</pre>

</blockquote>

### Actual behavior

```sh
gosec ./tmp/
```

```none
[gosec] 2022/06/16 20:14:40 Including rules: default
[gosec] 2022/06/16 20:14:40 Excluding rules: default
[gosec] 2022/06/16 20:14:40 Import directory: REMOVED
[gosec] 2022/06/16 20:14:41 Checking package: main
[gosec] 2022/06/16 20:14:41 Checking file: REMOVED/main.go
Results:


[REMOVED/main.go:10-14] - G112 (CWE-400): Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server (Confidence: LOW, Severity: MEDIUM)
    9: func main() {
  > 10:         fmt.Println(&http.Server{
  > 11:                 ReadTimeout:  1 * time.Second,
  > 12:                 WriteTimeout: 1 * time.Second,
  > 13:                 IdleTimeout:  1 * time.Second,
  > 14:         })
    15: }



Summary:
  Gosec  : dev
  Files  : 1
  Lines  : 15
  Nosec  : 0
  Issues : 1
```

**Comment by yunwei37** [comment]
I'm sorry about that, it seems to be my mistake...It can be fixed though the PR #825

**Comment by ccojocar** [comment]
fixed by https://github.com/securego/gosec/pull/825
